socket.c

```diff
@@ -87,23 +88,56 @@ void bind_to_telnet_port(char type, int *socketdescriptor)
 	}
 
 	bool bound = false;
+	// Try dual-stack socket
 	for(port = port_init; port <= (port_init + 20); port++)
 	{
-		serv_addr.sin_port = htons(port);
+		serv_addr.sin6_port = htons(port);
 		if(bind(*socketdescriptor, (struct sockaddr *) &serv_addr, sizeof(serv_addr)) < 0)
 		{
 			logg("Error on binding on port %i", port);
 		}
 		else
 		{
 			bound = true;
+			dualstack = true;
 			break;
 		}
 	}
```

Comment:
On a non-IPv6 system this would produce 20 lines of errors saying it can't bind to a port. Is there a quicker way to figure out if we have IPv6 access?

Reply:
I'm not sure how to detect if the system has a working `ipv6` interface. If we cannot find out how to do it, we could also change the error reporting (don't report on the Individual failures, but only once at the end when we couldn't find at least one port).